core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -400,7 +398,9 @@ private[kafka] class Processor(val id: Int,
 
   private val newConnections = new ConcurrentLinkedQueue[SocketChannel]()
   private val inflightResponses = mutable.Map[String, RequestChannel.Response]()
-  private val metricTags = Map("networkProcessor" -> id.toString).asJava
+  val metricTags = Map(
+    "protocol" -> securityProtocol.name, "listener" -> listenerName.value, "networkProcessor" -> id.toString
+    ).asJava
```

Comment:
I had the protocol as first one, as when using a treeview representation, allows for the listeners to be shown as children. 
In fact, protocol/listener/networkProcessor is the right order from the most generic to the most specific tag.